sycl/include/sycl/exception.hpp

```diff
@@ -70,6 +70,8 @@ class __SYCL_EXPORT SYCLCategory : public std::error_category {
 
 // Forward declare to declare as a friend in sycl::excepton.
 __SYCL_EXPORT pi_int32 get_pi_error(const exception &e);
+// TODO: Should it be exported at all?
+__SYCL_EXPORT exception set_pi_error(exception &&e, pi_int32 pi_err);
```

Comment:
Is there any problem if it's not exported? I'd rather not export if if there's no reason.

Reply:
That's essentially the same as https://github.com/intel/llvm/pull/14400/files#r1663304397. I'm not yet sure why it's been done the way it is, but I'd prefer to handle that after old SYCL 1.2 exceptions are removed as I don't want to block other PRs that are dependent on this one.

Reply:
That's okay, I'm not blocking the PR because of this.